packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,757 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  ReactElement,
+  memo,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  MRT_Cell,
+  MRT_DensityState,
+  MRT_Row,
+  MRT_RowData,
+  MRT_SortingState,
+  MRT_TableOptions,
+  MRT_RowSelectionState,
+  MRT_RowVirtualizer,
+  MRT_VisibilityState,
+  useMaterialReactTable,
+  MRT_TableContainer,
+} from "material-react-table";
+import {
+  ArrowDownIcon,
+  ArrowUnsortedIcon,
+  DragIndicatorIcon,
+  MoreIcon,
+} from "../icons.generated";
+import { densityValues, paginationTypeValues } from "./constants";
+import { DataTablePagination } from "./DataTablePagination";
+import { DataFilter, DataFilters } from "../labs/DataFilters";
+import {
+  DataTableRowActions,
+  DataTableRowActionsProps,
+} from "./DataTableRowActions";
+import { useRowReordering } from "./useRowReordering";
+import { DataTableSettings } from "./DataTableSettings";
+import { MenuButton, MenuButtonProps } from "../MenuButton";
+import { Box } from "../Box";
+import { DataTableRowSelectionState } from ".";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+import { useScrollIndication } from "./useScrollIndication";
+import styled from "@emotion/styled";
+import { DataTableEmptyState } from "./DataTableEmptyState";
+import { Callout } from "../Callout";
+import { t } from "i18next";
+
+export type DataTableGetDataType = {
+  page?: number;
+  resultsPerPage?: number;
+  search?: string;
+  filters?: DataFilter[];
+  sort?: MRT_SortingState;
+};
+
+export type DataTableOnReorderRowsType = {
+  rowId: string;
+  newRowIndex: number;
+};
+
+export type DataTableProps = {
+  /**
+   * The columns that make up the table
+   */
+  columns: MRT_TableOptions<MRT_RowData>["columns"];
+  /**
+   * The total number of rows in the table. Optional, because it's sometimes impossible
+   * to calculate. Used in table pagination to know when to disable the "next"/"more" button.
+   */
+  totalRows?: number;
+  /**
+   * The function to get the ID of a row
+   */
+  getRowId?: MRT_TableOptions<MRT_RowData>["getRowId"];
+  /**
+   * The initial density of the table. This is available even if the table density
+   * isn't changeable.
+   */
```

Comment:
What is density though? From this comment, I don't really understand what I need to provide here